stl/inc/flat_map

```diff
@@ -960,210 +886,530 @@ private:
+    }
+
+    template <class _MappedTy>
+    pair<iterator, bool> insert_or_assign(const key_type& _Key_val, _MappedTy&& _Mapped_val)
+        requires is_assignable_v<mapped_type&, _MappedTy> && is_constructible_v<mapped_type, _MappedTy>
+    {
+        return _Insert_or_assign(_Key_val, _STD forward<_MappedTy>(_Mapped_val));
+    }
+
+    template <class _MappedTy>
+    pair<iterator, bool> insert_or_assign(key_type&& _Key_val, _MappedTy&& _Mapped_val)
+        requires is_assignable_v<mapped_type&, _MappedTy> && is_constructible_v<mapped_type, _MappedTy>
+    {
+        return _Insert_or_assign(_STD move(_Key_val), _STD forward<_MappedTy>(_Mapped_val));
+    }
+
+    template <class _OtherKey, class _MappedTy>
+    pair<iterator, bool> insert_or_assign(_OtherKey&& _Key_val, _MappedTy&& _Mapped_val)
+        requires _Is_transparent_v<key_compare> && is_constructible_v<key_type, _OtherKey>
+              && is_assignable_v<mapped_type&, _MappedTy> && is_constructible_v<mapped_type, _MappedTy>
+    {
+        return _Insert_or_assign(_STD forward<_OtherKey>(_Key_val), _STD forward<_MappedTy>(_Mapped_val));
+    }
+
+    template <class _MappedTy>
+    iterator insert_or_assign(const_iterator _Position, const key_type& _Key_val, _MappedTy&& _Mapped_val)
+        requires is_assignable_v<mapped_type&, _MappedTy> && is_constructible_v<mapped_type, _MappedTy>
+    {
+        return this->_Emplace_hint<true>(_Position, _Key_val, _STD forward<_MappedTy>(_Mapped_val));
+    }
+
+    template <class _MappedTy>
+    iterator insert_or_assign(const_iterator _Position, key_type&& _Key_val, _MappedTy&& _Mapped_val)
+        requires is_assignable_v<mapped_type&, _MappedTy> && is_constructible_v<mapped_type, _MappedTy>
+    {
+        return this->_Emplace_hint<true>(_Position, _STD move(_Key_val), _STD forward<_MappedTy>(_Mapped_val));
+    }
+
+    template <class _OtherKey, class _MappedTy>
+    iterator insert_or_assign(const_iterator _Position, _OtherKey&& _Key_val, _MappedTy&& _Mapped_val)
+        requires _Is_transparent_v<key_compare> && is_constructible_v<key_type, _OtherKey>
+              && is_assignable_v<mapped_type&, _MappedTy> && is_constructible_v<mapped_type, _MappedTy>
+    {
+        return this->_Emplace_hint<true>(
+            _Position, _STD forward<_OtherKey>(_Key_val), _STD forward<_MappedTy>(_Mapped_val));
+    }
+
 private:
-    using _MyBase = _Flat_Map_Base<_Key, _Mapped, _Compare, _KeyContainer, _MappedContainer, true,
-        flat_multimap<_Key, _Mapped, _Compare, _KeyContainer, _MappedContainer>>;
+    using _MyBase::_Erase_if;
 
-public:
-    using _MyBase::_MyBase;
-    using _MyBase::operator=;
+    template <class _KTy, class _MTy, class _Comp, class _KeyCont, class _MappedCont, class _Pred>
+    friend typename flat_map<_KTy, _MTy, _Comp, _KeyCont, _MappedCont>::size_type erase_if(
+        flat_map<_KTy, _MTy, _Comp, _KeyCont, _MappedCont>&, _Pred);

```

Comment:
`typename` shouldn't be necessary in the return type here (checked on Compiler Explorer). Occurs below (in the function definition, and twice more for `flat_multimap`, I believe).

Reply:
Hmm... this was pointed (for `flat_(multi_)set`) before (https://github.com/microsoft/STL/pull/3833#discussion_r1241279116).
I'd like to open an editorial for C++ working draft, but it's a bit unfornate that WG21-P2150 is already handling this and the paper is stuck.